docs/architecture/decisions/0002-docker-for-development.md

```diff
@@ -0,0 +1,25 @@
+# 2. Docker for development
+
+Date: 2022-08-09
+
+## Status
+
+Accepted
+
+## Context
+
+The [Federal Audit Clearinghouse](https://github.com/GSA-TTS/FAC) project is a similar project to ours, from which we are taking many examples for reuse. FAC uses Docker. Docker is widely accepted at TTS currently and in the development community at large.
+
+Other options include using virtual machines or installing dependencies directly onto the host system. The first is not widely used at TTS; the second tends to cause problems with dependency management.
+
+## Decision
+
+To include a Dockerfile and docker-compose file.
+
+## Consequences
+
+We intend to deploy using containerized images; therefore, use of Docker on local will give the closest approximation to higher environments.
+
+The project may be perceived as less friendly to outsiders.
```

Comment:
What does this mean? Why so?

Reply:
After giving some thought, I've removed this line.

It has to do with history and culture in ways that are difficult to summarize in a brief paragraph. There's a quote from Brendan Eich in that [JavaScript article](https://devclass.com/2022/08/04/retire_javascript_says-json-creator-douglas-crockford/) that was recently shared in Slack which came close: "I went in to do … a programming language for HTML, for web designers and programmers to use, embedded directly in the web page … not like Java which was a professional’s language where you would run real code with type declarations, and you’d have to write in a way that compiled."